Sources/AppcuesKit/Presentation/Debugger/ScreenCapturing/CaptureToastView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  CaptureToastView.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 3/8/23.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import UIKit
+
+internal class CaptureToastView: UIView {
+
+    var onRetry: (() -> Void)?
+
+    private var stackView: UIStackView = {
+        let view = UIStackView()
+        view.axis = .horizontal
+        view.alignment = .center
+        view.spacing = 8.0
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private var messageLabel: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        return label
+    }()
+
+    private var retryButton: UIButton = {
+        let button = UIButton()
+        button.setTitle("Try again", for: .normal)
+        button.titleLabel?.font = .systemFont(ofSize: 14)
+        button.layer.borderWidth = 1.0
+        button.layer.borderColor = UIColor.white.cgColor
+        button.layer.cornerRadius = 6.0
+        return button
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        layer.cornerRadius = 6.0
+
+        stackView.addArrangedSubview(messageLabel)
+        stackView.addArrangedSubview(retryButton)
+
+        addSubview(stackView)
+
+        NSLayoutConstraint.activate([
+            stackView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 16),
+            stackView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -16),
+            stackView.topAnchor.constraint(equalTo: topAnchor, constant: 12),
+            stackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12),
```

Comment:
This is fine, but in case you didn't know, you can set `layoutMargins` on the stack view and then set `stackView.isLayoutMarginsRelativeArrangement = true`. Which approach you take really is just dependent on how you visualize the view in your mind. One slight benefit of the layout margins is not having the extra cognitive load of "why is this constant value negative again?"

Reply:
nice, yeah that is probably a cleaner route here, will try it out, thanks.